lib/modules/redirect.js

```diff
@@ -0,0 +1,116 @@
+/* @flow */
+
+import { Module } from '../core/module';
+import { isEnabled, setEnabled, getOptionsValues } from '../core/modules';
+import { Permissions, Redirection } from '../environment';
+import { showConfigOptions } from './settingsConsole';
+
+export const module: Module<*> = new Module('redirect');
+
+module.moduleName = 'redirectName';
+module.category = 'usersCategory';
+module.description = 'redirectDesc';
+module.disabledByDefault = true;
+module.keywords = ['redirect'];
+
+module.options = {
+	fromProfileLandingPage: {
+		title: 'redirectFromProfileLandingPageTitle',
+		description: 'redirectFromProfileLandingPageDesc',
+		keywords: ['legacy', 'overview'],
+		type: 'enum',
+		value: 'none',
+		values: [{
+			name: 'Do nothing',
+			value: 'none',
+		}, {
+			name: 'Overview (legacy)',
+			value: 'overview',
+		}, {
+			name: 'Comments',
+			value: 'comments',
+		}, {
+			name: 'Submitted (legacy)',
+			value: 'submitted',
+		}, {
+			name: 'Gilded',
+			value: 'gilded',
+		}, {
+			name: 'Custom',
+			value: 'custom',
+		}],
+	},
+	customFromProfileLandingPage: {
+		dependsOn: options => options.fromProfileLandingPage.value === 'custom',
+		title: 'redirectCustomFromProfileLandingPageTitle',
+		description: 'redirectCustomFromProfileLandingPageDesc',
+		type: 'text',
+		value: '',
+	},
+	fromSubredditFrontPage: {
+		title: 'redirectFromSubredditFrontPageTitle',
+		description: 'redirectFromSubredditFrontPageDesc',
+		keywords: ['subreddit', 'front'],
+		type: 'enum',
+		value: 'none',
+		values: [{
+			name: 'Do nothing',
+			value: 'none',
+		}, {
+			name: 'New',
+			value: 'new',
+		}, {
+			name: 'Rising',
+			value: 'rising',
+		}, {
+			name: 'Controversial',
+			value: 'controversial',
+		}, {
+			name: 'Top',
+			value: 'top',
+		}, {
+			name: 'Gilded',
+			value: 'gilded',
+		}, {
+			name: 'Wiki',
+			value: 'wiki',
+		}, {
+			name: 'Custom',
+			value: 'custom',
+		}],
+	},
+	customFromSubredditFrontPage: {
+		dependsOn: options => options.fromSubredditFrontPage.value === 'custom',
+		title: 'redirectCustomFromSubredditFrontPageTitle',
+		description: 'redirectCustomFromSubredditFrontPageDesc',
+		type: 'text',
+		value: '',
+	},
+};
+
+module.beforeLoad = updateRedirectState;
+
+module.onToggle = enabled => {
+	if (enabled) {
+		Permissions.request(['webRequestBlocking'])
```

Comment:
I could add an "enableModule" function (and optionally a simple disableModule function for symmetry) to the settingsConsole module that
1. checks what permissions the module we're about to enable needs
2. requests permissions from the user as necessary
3. only actually enabled the module when the required permissions are granted

Is this what you had in mind?

Reply:
Done in commit #e7e8f9a. The toggleHandler fails silently (i.e. throws an error that the user doesn't see) if the required permissions are not granted. Is this ok, since it should be obvious (?) to the user that the permissions we're asking for are required to enable the module, or should I be showing a notification?

Reply:
I think it's obvious, but we should probably create an notification to explain why permissions are requested. Maybe like showImages does it?
https://github.com/honestbleeps/Reddit-Enhancement-Suite/blob/256c10a2c035fb20614e7dc6114b99fd123c2269/lib/modules/showImages.js#L859-L869

Reply:
Changes in commit #60dee98 allow modules to control the notification displayed when requesting extra permissions

But:
Turns out chrome [doesn't](https://developer.chrome.com/apps/permission_warnings#permissions_with_warnings) show the user a warning prompt when an extension requests the `webRequestBlocking` permission via `chrome.permissions`. Instead, the permission is granted silently (I've tested this)

Since users might find it strange that RES is explaining why it needs extra permissions when there's no prompt, I've made the permission-notification optional, and am not currently showing one for the Redirects module

Reply:
There's no prompt even if the permission is being requested for the first time, not just when it is being requested after already having been granted